Updates for the api section docs

jonniebigodes opened the pull request:
This pull request starts to address some of the necessary items that need to be fixed for this section.

What was done:
- Updated the links to incorporate the extension `.md`.
- Some images that were being pulled in directly from the repo are now local.
- Updated reference of `props table` to `args table` and the link is pointing it.


Feel free to provide feedback

jonniebigodes committed:
links updated and images added

tmeasday commented on docs/api/addons.md:
```diff
@@ -258,6 +258,9 @@ export const parameters = {
+TODO: vet this documentation link if it's correct.
+</div>
 
 For more information on presets, see the [presets docs](../presets/introduction/index).
```
Yeah I don't think this link exists

tmeasday reviewed (approved):
Looks good

jonniebigodes committed:
links updated and break down subpages

jonniebigodes committed:
updated toc with new changes

jonniebigodes commented:
with the latest commit the api section was broken down into smaller subpages and the content was moved over and links updated.

The `toc.js` file was updated to match the new changes.

tmeasday commented on docs/api/stories.md:
```diff
@@ -2,516 +2,10 @@
-
-Be sure to update your Storybook config file to load `.stories.mdx` stories, as per the [`addon-docs` installation instructions](https://github.com/storybookjs/storybook/tree/master/addons/docs#installation).
-
-### ArgTypes
```
I think this section got dropped by accident.

jonniebigodes replied:
Based on the feedback that i got, there was a need to split it to smaller subpages that target those items. One for mdx, one for csf, one for storiesOf.

@domyen care to chime in

domyen replied:
Yea, we're splitting these pages up so that we can link to individual pages.
![image](https://user-images.githubusercontent.com/263385/89462381-4e9a5180-d73b-11ea-8458-46484500384d.png)

domyen replied:
Ignore the StoriesOf API, that's no longer present in the docs

tmeasday commented on docs/api/storiesof.md:
```diff
@@ -0,0 +1,93 @@
+---
```
Was this intentionally re-added? I was thinking we were going to move this to a `ADVANCED.md` or something.

jonniebigodes committed:
cli converted to table and included docs flag

jonniebigodes commented:
with the latest commit i've added the `--docs` flag as well as it was missing from the cli options page. And converted it to a table for better readability.

jonniebigodes committed:
adds missing link to the writing docs page.

jonniebigodes committed:
updated per feedback provided.

jonniebigodes commented:
with the latest commit i've updated the documentation per feedback. 

Once this is vetted, i'll resolve the `toc.js` conflicts and we get it merged.

Feel free to provide feedback.

tmeasday commented on docs/api/csf.md:
```diff
@@ -0,0 +1,203 @@
+
+<div class="aside"
+
+If you are writing stories in the older `storiesOf()` syntax, you can find documentation in an [advanced README](https://github.com/storybookjs/storybook/tree/next/lib/core/ADVANCED.md).
```
Can we make these links relative I wonder?

jonniebigodes replied:
converted the link to be relative. Once the file is created it should fall into place.

tmeasday reviewed (approved):
LGTM

jonniebigodes committed:
minor tweaks to the toc and csf doc.

jonniebigodes committed:
Merge branch '6.0-docs' into api_section_updates

jonniebigodes commented:
merging as is, further improvements will be made with follow up pull requests.
